content/articles/building-a-simple-translation-app-using-python-for-beginners/index.md

```diff
@@ -0,0 +1,171 @@
+
+To do this, let's type the following code:
+
+```python
+    translation_pipeline = pipeline ('translation_en_to_fr')
+```
+Let's try it out. So, for example, if I wanted to translate the sentence, "I can't wait to watch Money Heist", we can do it like this:
+
+```python
+    translation_pipeline ('I cannot wait to watch Money Heist')
+```
+This will be the output:
+
+```python
+
+    [{'translation_text': 'Je ne peux attendre de regarder Money Heist'}]
+
+```
+
+Amazing right? The translation pipeline is translating our text from English to French. 
+
+Lastly, we are going to create a function to wrap up all these and display it on a beautiful user interface. We'll be leveraging Gradio for this task.
+
+### Building the web app for interaction using Gradio
+This web app allows a user to pass through some text and have it translated.
+
+```python
+    def translate_transformers(from_text):
+    translated_results = translation_pipeline(from_text)
+    return translated_results[0]['translation_text']
+```
+
+We have created a function called `translate_transformers()`. We then pass through the text to our translate transformer pipeline. We are storing the results of the translation in a variable called `translated_results`. The function then grabs the first value in our list [0] and returns the translated text.
+
+Now, all that's left to do is wrap this up inside the Gradio function.
+
+We create the gradio interface by writing following pieces of code:
+
+```python
+interface = gr.Interface(fn=translate_transformers, 
+                        inputs=gr.inputs.Textbox(lines=2, placeholder='Text to translate'),
+                        outputs='text')
+```
+It's amazing how quickly you can create this interface, with only a single line of code. We've passed in a few arguments. `fn` is where we pass in our function which is `translate_transformers`. We pass in an `input` type of Textbox with 2 lines and a placeholder. You can change the number of lines and placeholders in your project without any issues. The last argument states the output type. For our case, it's `text`. 
+
+Let's go ahead and launch it. We use the `launch()` method to achieve this. 
+
+```python
+interface.launch()
+```
+This command will run our gradio app.
+
+![Gradio app](/engineering-education/building-a-simple-translation-app-using-python-for-beginners/gradio-app.png)
+
+*[Image Source: Colab](https://colab.research.google.com/drive/1MMgN89Xn7fz5zVh1XCuLd66ZBIoZ70DZ#scrollTo=62a4vvW_dLsx)*
+
```

Comment:
You can remove the source of the image as it is your own and not an external source.